pkg/sdk/data_types.go

```diff
@@ -112,3 +118,59 @@ func IsStringType(_type string) bool {
 		strings.HasPrefix(t, "NVARCHAR") ||
 		strings.HasPrefix(t, "NCHAR")
 }
+
+// ParseNumberDataTypeRaw extracts precision and scale from the raw number data type input.
+// It returns defaults if it can't parse arguments, data type is different, or no arguments were provided.
+// TODO [SNOW-1348103 or SNOW-1348106]: visit with functions and procedures rework
+func ParseNumberDataTypeRaw(rawDataType string) (int, int) {
+	r := util.TrimAllPrefixes(strings.TrimSpace(strings.ToUpper(rawDataType)), DataTypeNumberSynonyms...)
+	r = strings.TrimSpace(r)
+	if strings.HasPrefix(r, "(") && strings.HasSuffix(r, ")") {
+		parts := strings.Split(r[1:len(r)-1], ",")
+		switch l := len(parts); l {
+		case 1:
+			precision, err := strconv.Atoi(strings.TrimSpace(parts[0]))
+			if err == nil {
+				return precision, DefaultNumberScale
+			} else {
+				logging.DebugLogger.Printf(`[DEBUG] Could not parse number precision "%s", err: %v`, parts[0], err)
+			}
+		case 2:
+			precision, err1 := strconv.Atoi(strings.TrimSpace(parts[0]))
+			scale, err2 := strconv.Atoi(strings.TrimSpace(parts[1]))
+			if err1 == nil && err2 == nil {
+				return precision, scale
+			} else {
+				logging.DebugLogger.Printf(`[DEBUG] Could not parse number precision "%s" or scale "%s", errs: %v, %v`, parts[0], parts[1], err1, err2)
+			}
+		default:
+			logging.DebugLogger.Printf("[DEBUG] Unexpected length of number arguments")
```

Comment:
Nit: I would add more description, at least a function name because this is too generic. In other lines as well.

Reply:
Added a function name temporarily.

FYI, we have a plan to rework logging inside the provider and the SDK as part of SNOW-1325306. The idea was, to have a logger that adds the code place on its own, so such descriptions won't be necessary and will need to be altered to reduce the noise.